src/Compilers/VisualBasic/Portable/Binding/Binder_ObjectInitializer.vb

```diff
@@ -905,6 +905,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                                           callerInfoOpt:=topLevelInitializer)
                 invocation.SetWasCompilerGenerated()
 
+                If (invocation.Kind = BoundKind.LateInvocation) Then
+                    invocation = DirectCast(invocation, BoundLateInvocation).SetLateBoundAccessKind(LateBoundAccessKind.Call)
```

Comment:
>invocation = DirectCast(invocation, BoundLateInvocation).SetLateBoundAccessKind(LateBoundAccessKind.Call) [](start = 20, length = 105)

Could you please confirm (by examining IL) that we keep calling the same VB runtime helper after this change and also compare against native compiler?

Reply:
This change brings us inline with the native compiler, looks like we had a bug here. Previous to this change, we were calling `[Microsoft.VisualBasic]Microsoft.VisualBasic.CompilerServices.NewLateBinding::LateGet`, whereas the native compiler called: `[Microsoft.VisualBasic]Microsoft.VisualBasic.CompilerServices.NewLateBinding::LateCall`. With this change, we also call `LateCall`.

---
In reply to: [190973564](https://github.com/dotnet/roslyn/pull/27126#discussion_r190973564) [](ancestors = 190973564)